extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -114,32 +117,40 @@ public OpenCensusProtobufInputRowParser withParseSpec(ParseSpec parseSpec)
         resourceLabelPrefix);
   }
 
+
+  private interface LabelContext
+  {
+    void addRow(long millis, String metricName, Object value);
+  }
+
   @Override
   public List<InputRow> parseBatch(ByteBuffer input)
   {
-
-    Metric metric;
+    final Metric metric;
     try {
-      metric = Metric.parseFrom(ByteString.copyFrom(input));
+      metric = Metric.parseFrom(input);
     }
     catch (InvalidProtocolBufferException e) {
       throw new ParseException(e, "Protobuf message could not be parsed");
     }
 
     // Process metric descriptor labels map keys.
-    List<String> descriptorLabels = metric.getMetricDescriptor().getLabelKeysList().stream()
-        .map(s -> this.metricLabelPrefix + s.getKey())
-        .collect(Collectors.toList());
+    List<String> descriptorLabels = new ArrayList<>(metric.getMetricDescriptor().getLabelKeysCount());
+    for (LabelKey s : metric.getMetricDescriptor().getLabelKeysList()) {
+      descriptorLabels.add(this.metricLabelPrefix + s.getKey());
+    }
```

Comment:
I prefer the original `stream` implementation :stuck_out_tongue: 
Guess this change improves performance on the microbenchmark?

Reply:
I did try an implementation where I used Streams everywhere, including replacing the nested for-loops further down with `streams.flatMap(...)` While I still got perf some improvement over the original code, it was not as good as when I removed Streams altogether.

Reply:
:+1: 

Since this is performance-sensitive code, I won't object :slightly_smiling_face: